src/EditorFeatures/Core/EditorConfigSettings/Updater/SettingsUpdateHelper.cs

```diff
@@ -167,24 +190,47 @@ internal static partial class SettingsUpdateHelper
             TextLine? lastValidSpecificHeader = null;
             TextLine? lastValidSpecificHeaderSpanEnd = null;
 
-            var textChange = new TextChange();
+            var textChanges = new List<TextChange>();
+            var foundOptionName = false;
             foreach (var curLine in editorConfigText.Lines)
             {
                 var curLineText = curLine.ToString();
-                if (s_optionEntryPattern.IsMatch(curLineText))
+                if (IsOption(curLineText))
                 {
-                    var groups = s_optionEntryPattern.Match(curLineText).Groups;
-                    var (untrimmedKey, key, value, severity, comment) = GetGroups(groups);
-
+                    var (untrimmedKey, key, value, severity, comment) = GetOptionParts(curLineText);
                     // Verify the most recent header is a valid header
-                    if (IsValidHeader(mostRecentHeader, lastValidHeader) &&
-                        string.Equals(key, optionName, StringComparison.OrdinalIgnoreCase))
+                    if (IsValidHeader(mostRecentHeader, lastValidHeader))
                     {
-                        // We found the rule in the file -- replace it with updated option value.
-                        textChange = new TextChange(curLine.Span, $"{untrimmedKey}={optionValue}{comment}");
+                        // Verify the option name matches
+                        if (string.Equals(key, optionName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var sb = new StringBuilder($"{optionName}={optionValue}\r\n");
```

Comment:
should we be hardcoding \r\n in these guys? my preference woudl at least be that that's a constant defined somewhere in this file, so we can tweak this at will, or use an env variable, or allow this to be passed in.

Reply:
yep I have it in a todo to actually try and get the users newline preferences for this file type if possible. In the meantime the codefix for editorconfig has been writing out `\r\n` since mid 2019.